Sources/MessagingInApp/MessagingInAppConfigOptions.swift

```diff
@@ -1,2 +1,59 @@
+import CioInternalCommon
+
+/**
+ Configuration options for in-app messaging module
+
+ Example usage:
+ ```
+ // create a new instance
+ let inAppMessagingConfig = MessagingInAppConfigOptions.Factory.create(siteId: siteId, region: .US)
+ ```
+ */
+// TODO: [CDP] Update docs to after module initialization changes for clarity on its usage
 public struct MessagingInAppConfigOptions {
```

Comment:
```suggestion
public struct MessagingInAppConfigOptions {
    public private(set) var siteId: String
    public private(set) var region: Region
    
    public var foo: Bar = true
    
    internal static func getInstance(siteId: String, region: Region) -> MessagingInAppConfigOptions {
      return MessagingInAppConfigOptions(siteId: siteId, region: Region)
    }
}
```

Not a blocker. IMO, the factory and keys enum feel like boilerplate to me and not needed? Swift structs are pretty powerful where we may not need to have them at all? 

Update after more review: You can disregard this comment if you wish. I see that this is how `main` currently behaves and you're following the same design pattern. Therefore, we can change this behavior some time in the future if we wish.

Reply:
Correct. I just followed the practice we're using in `main`. I'm open to going without Factory too, but for this PR, let's stick with the Factory approach to maintain consistency.